lib/Backends/CPU/libjit/libjit_matmul.cpp

```diff
@@ -83,14 +188,19 @@ void libjit_matmul_inner(int m, int n, int k, const float *a, int lda,
   // perfectly-tiled portion, which we handly with a 4x16 dot-product kernel.
   // The ragged edges are (ideally) less critical, so we handle them with a call
   // to a general matrix-multiplication for odd sizes.
-  for (int j = 0; j < n - nr + 1; j += nr) {
-    for (int i = 0; i < m - mc + 1; i += mc) {
-      libjit_matmul_dot<regsA, regsB>(k, &A(i, 0), lda, &B(0, j), ldb, &C(i, j),
-                                      ldc);
-    }
+  float packedA[m * k] __attribute__((aligned(64)));
```

Comment:
@bertmaher Nit: It looks like you reserve memory for `packedA` even if `pack` is false. You do the same for `packedB` below as well. I wonder if the code can be slightly re-arranges to reserve the memory for packed matrices only when really needed. WDYT?

Reply:
I'm a bit uncertain.  I think `A` could be done trivially, but `B` is a bit trickier.  (Obviously I could specialize the whole code path "by hand", but I rather wanted to avoid some code duplication.)